src/plugins_extra/selectZoom.ts

```diff
@@ -95,19 +99,20 @@ export class SelectZoom {
             return;
         const p = this.getPoint(ev);
 
-        if (this.options.enableX) {
-            const x = Math.min(this.start.p.x, p.x);
-            const w = Math.abs(this.start.p.x - p.x);
+        const x = Math.min(this.start.p.x, p.x);
+        const w = Math.abs(this.start.p.x - p.x);
+        const y = Math.min(this.start.p.y, p.y);
+        const h = Math.abs(this.start.p.y - p.y);
+
+        if (this.options.enableX && ((w >= h) || (w >= this.options.thresholdX))) {
```

Comment:
```suggestion
        if (this.options.enableX && (w >= this.options.thresholdX)) {
```
I think this condition `w >= h` is confusing. The preview is shown, but the zoom will not be executed.

Reply:
You are right, there was a huge oversight on my part since I only tested the feature with very small thresholds...

The application of the zoom did not match the visual indicator if the height and/or width were below their respective thresholds. But the problem was not in the `onMouseMove()` event as you suggested. Instead I adapted the `onMouseUp()` event to work exactly like the zoom indicator suggests.

Reply:
And regarding the documentation. I could add a section that introduces the `SelectZoomPlugin` just after the `TimeChartZoomPlugin` [here](https://github.com/huww98/TimeChart?tab=readme-ov-file#zoom-options)?

Reply:
> Instead I adapted the onMouseUp() event to work exactly like the zoom indicator suggests.

I personally do not like this. As you can see, the if statement becomes rather complex, and can be hard to reason about. But fine if you insist and it is properly documented.

> And regarding the documentation. I could add a section that introduces the SelectZoomPlugin just after the TimeChartZoomPlugin here?

You may add a link there. But I believe the main documentation should live at https://github.com/huww98/TimeChart/blob/master/src/plugins_extra/README.md#select-zoom

Reply:
I tried to increase the readability of the if statement and I introduced private properties for the boolean flags that decide whether an axis should be selected or not. This ensures that the same rules are applied in the events onMouseMove and onMouseUp.
I also added two inputs for the threshold in the select-zoom demo. I have never worked with forms so I just tried to copy the way you handled the checkboxes.

Do you want me to add any inline comments to explain the thresholds? And do you want me to also add documentation to the link you posted in your last comment? Or is it sufficient to have the threshold inputs in the demo?